wandb/util.py

```diff
@@ -146,22 +167,55 @@ def sentry_reraise(exc: Any) -> None:
 
 
 def sentry_set_scope(
-    process_context: Optional[str],
-    entity: Optional[str],
-    project: Optional[str],
-    email: Optional[str] = None,
-    url: Optional[str] = None,
+    settings_dict: Optional[
+        Union[
+            "wandb.sdk.wandb_settings.Settings",
+            "wandb.sdk.internal.settings_static.SettingsStatic",
+        ]
+    ] = None,
+    process_context: Optional[str] = None,
 ) -> None:
     # Using GLOBAL_HUB means these tags will persist between threads.
     # Normally there is one hub per thread.
+
+    # Normally, we pull all tags from the passed-in settings object.
+    # However, we also allow callers of this func to pass in arbitrary key-value
+    # pairs as tags.
+    args = locals()
```

Comment:
there is a python oddity with locals(),  the workaround is using: dict(locals())